server.js

```diff
@@ -1,43 +1,52 @@
 import express from 'express'
+import fs from 'fs';
 import React from 'react'
 import { renderToString } from 'react-dom/server';
+import { Provider } from 'react-redux'
+import { StaticRouter as Router } from 'react-router-dom';
+import { matchRoutes } from 'react-router-config'
+
 import App from './src/App';
+import { configureStore } from './src/configureStore';
+import routes from './src/routeConfig';
 
 const app = express();
 const port = 3000;
 
-app.use('/build/', express.static('static'));
+app.use(express.static(__dirname + '/build'));
 
 app.use(handleRender);
 
 function handleRender(req, res) {
-    const html = renderToString(
-        <App />
-    );
+    const store = configureStore();
+    const context = {};
+    const branch = matchRoutes(routes, req.path);
 
-    res.send(renderFullPage(html, { data: 'data' }))
-}
+    const stateToResolve = branch.map(({ route, match}) => {
+        return route.component.prepareState(store, match, req.query);
+    });
+
+    return Promise.all(stateToResolve).then(() => {
+        fs.readFile('./build/template.html', 'utf8', (err, data) => {
+            if (err) throw err;
```

Comment:
This will make your execution environment catch error. Please try to handle it properly: your function returns promise, it should return reject on error.